shm/Map.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <vector>
+#include "island.hpp"
+#include <random>
+#include <algorithm>
+#include <utility>
+#include <iostream>
+#include "coordinates.hpp"
+
+constexpr size_t COORDINATE_MIN = 0;
+constexpr size_t COORDINATE_MAX = 99;
+constexpr size_t ISLANDS_COUNT = 10;
```

Comment:
move to class. If this value is needed only in .cpp file then move to the anonymous namespace in .cpp file.